### PR TITLE
Feature/3568 xml quoted attachement

### PR DESCRIPTION
**Pull request by TGazica**
### 🎯 Goal

Add support to XML SDK for custom quoted attachments.

### 🛠 Implementation details

Added `QuotedAttachmentFactory` which is used to generate custom preview for attachments inside quoted messages. Also fixed the `MessageReplyView`.

### 🎨 UI Changes

| Before | After |
| --- | --- |
| Screenshot_20220525_131847 | quoted_attachment_final |

### 🧪 Testing

<details>
  <summary>Patch</summary>

```
Index: stream-chat-android-ui-components-sample/src/main/res/layout/view_quoted_date_attachemnt.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/stream-chat-android-ui-components-sample/src/main/res/layout/view_quoted_date_attachemnt.xml b/stream-chat-android-ui-components-sample/src/main/res/layout/view_quoted_date_attachemnt.xml
new file mode 100644
--- /dev/null	(date 1653477618607)
+++ b/stream-chat-android-ui-components-sample/src/main/res/layout/view_quoted_date_attachemnt.xml	(date 1653477618607)
@@ -0,0 +1,46 @@
+<?xml version="1.0" encoding="utf-8"?>
+<merge xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    tools:parentTag="android.widget.FrameLayout"
+    >
+
+    <androidx.cardview.widget.CardView
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        app:cardBackgroundColor="#00FF00"
+        app:cardCornerRadius="16dp"
+        app:cardElevation="0dp"
+        >
+
+        <LinearLayout
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:gravity="center"
+            android:orientation="vertical"
+            >
+
+            <ImageView
+                android:layout_width="24dp"
+                android:layout_height="24dp"
+                android:layout_margin="4dp"
+                android:src="@drawable/ic_calendar"
+                tools:ignore="ContentDescription"
+                />
+
+            <TextView
+                android:id="@+id/dateTextView"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_marginStart="8dp"
+                android:layout_marginEnd="8dp"
+                android:layout_marginBottom="8dp"
+                android:gravity="center"
+                android:textColor="#000000"
+                tools:text="1982\nMay 21"
+                />
+
+        </LinearLayout>
+
+    </androidx.cardview.widget.CardView>
+</merge>
\ No newline at end of file
Index: stream-chat-android-ui-components-sample/src/main/res/layout/item_date_attachment.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/stream-chat-android-ui-components-sample/src/main/res/layout/item_date_attachment.xml b/stream-chat-android-ui-components-sample/src/main/res/layout/item_date_attachment.xml
new file mode 100644
--- /dev/null	(date 1653477618607)
+++ b/stream-chat-android-ui-components-sample/src/main/res/layout/item_date_attachment.xml	(date 1653477618607)
@@ -0,0 +1,31 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.cardview.widget.CardView xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:layout_margin="8dp"
+    app:cardBackgroundColor="#00FF00"
+    app:cardCornerRadius="16dp"
+    app:cardElevation="0dp"
+    >
+
+    <ImageView
+        android:layout_width="24dp"
+        android:layout_height="24dp"
+        android:layout_gravity="center_vertical"
+        android:layout_margin="8dp"
+        android:src="@drawable/ic_block"
+        tools:ignore="ContentDescription"
+        />
+
+    <TextView
+        android:id="@+id/dateTextView"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_gravity="center_vertical"
+        android:layout_marginStart="40dp"
+        android:textColor="#000000"
+        />
+
+</androidx.cardview.widget.CardView>
Index: stream-chat-android-ui-components-sample/src/main/res/drawable/ic_calendar.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/stream-chat-android-ui-components-sample/src/main/res/drawable/ic_calendar.xml b/stream-chat-android-ui-components-sample/src/main/res/drawable/ic_calendar.xml
new file mode 100644
--- /dev/null	(date 1653477618607)
+++ b/stream-chat-android-ui-components-sample/src/main/res/drawable/ic_calendar.xml	(date 1653477618607)
@@ -0,0 +1,10 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="24dp"
+    android:height="24dp"
+    android:viewportWidth="24"
+    android:viewportHeight="24"
+    android:tint="#000000">
+  <path
+      android:fillColor="#000000"
+      android:pathData="M19,4h-1V2h-2v2H8V2H6v2H5C3.89,4 3.01,4.9 3.01,6L3,20c0,1.1 0.89,2 2,2h14c1.1,0 2,-0.9 2,-2V6C21,4.9 20.1,4 19,4zM19,20H5V10h14V20zM9,14H7v-2h2V14zM13,14h-2v-2h2V14zM17,14h-2v-2h2V14zM9,18H7v-2h2V18zM13,18h-2v-2h2V18zM17,18h-2v-2h2V18z"/>
+</vector>
Index: stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/ChatFragment.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/ChatFragment.kt b/stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/ChatFragment.kt
--- a/stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/ChatFragment.kt	(revision 627e8513cd244917129896a3258b0781a6139562)
+++ b/stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/ChatFragment.kt	(date 1653477618607)
@@ -18,34 +18,54 @@
 
 package io.getstream.chat.ui.sample.feature.chat
 
+import android.content.Context
 import android.os.Bundle
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import android.widget.FrameLayout
 import androidx.activity.OnBackPressedCallback
+import androidx.cardview.widget.CardView
 import androidx.fragment.app.Fragment
 import androidx.fragment.app.viewModels
 import androidx.navigation.fragment.findNavController
 import androidx.navigation.fragment.navArgs
 import com.getstream.sdk.chat.utils.extensions.getCreatedAtOrThrow
+import com.getstream.sdk.chat.utils.extensions.inflater
 import com.getstream.sdk.chat.viewmodel.MessageInputViewModel
 import com.getstream.sdk.chat.viewmodel.messages.MessageListViewModel
+import com.google.android.material.datepicker.MaterialDatePicker
 import io.getstream.chat.android.client.errors.ChatNetworkError
+import io.getstream.chat.android.client.models.Attachment
 import io.getstream.chat.android.client.models.Flag
 import io.getstream.chat.android.client.models.Message
 import io.getstream.chat.android.client.utils.Result
+import io.getstream.chat.android.core.internal.InternalStreamChatApi
 import io.getstream.chat.android.livedata.utils.EventObserver
+import io.getstream.chat.android.ui.message.input.attachment.selected.internal.BaseSelectedCustomAttachmentViewHolder
+import io.getstream.chat.android.ui.message.input.attachment.selected.internal.SelectedCustomAttachmentViewHolderFactory
 import io.getstream.chat.android.ui.message.input.viewmodel.bindView
 import io.getstream.chat.android.ui.message.list.DeletedMessageListItemPredicate
+import io.getstream.chat.android.ui.message.list.adapter.MessageListListenerContainer
+import io.getstream.chat.android.ui.message.list.adapter.viewholder.attachment.AttachmentFactory
+import io.getstream.chat.android.ui.message.list.adapter.viewholder.attachment.InnerAttachmentViewHolder
+import io.getstream.chat.android.ui.message.list.adapter.viewholder.attachment.QuotedAttachmentFactory
 import io.getstream.chat.android.ui.message.list.header.viewmodel.MessageListHeaderViewModel
 import io.getstream.chat.android.ui.message.list.header.viewmodel.bindView
 import io.getstream.chat.android.ui.message.list.viewmodel.bindView
 import io.getstream.chat.android.ui.message.list.viewmodel.factory.MessageListViewModelFactory
 import io.getstream.chat.ui.sample.common.navigateSafely
 import io.getstream.chat.ui.sample.databinding.FragmentChatBinding
+import io.getstream.chat.ui.sample.databinding.ItemDateAttachmentBinding
+import io.getstream.chat.ui.sample.databinding.ItemDateAttachmentPreviewBinding
+import io.getstream.chat.ui.sample.databinding.ViewQuotedDateAttachemntBinding
 import io.getstream.chat.ui.sample.feature.common.ConfirmationDialogFragment
 import io.getstream.chat.ui.sample.util.extensions.useAdjustResize
+import java.text.DateFormat
+import java.text.SimpleDateFormat
 import java.util.Calendar
+import java.util.Date
+import java.util.Locale
 
 class ChatFragment : Fragment() {
 
@@ -151,6 +171,31 @@
             }
             binding.messageListView.setMessageEditHandler(::postMessageToEdit)
         }
+
+        // Create an instance of a date picker dialog
+        val datePickerDialog = MaterialDatePicker.Builder
+            .datePicker()
+            .build()
+
+        // Add an attachment to the message input when the user selects a date
+        datePickerDialog.addOnPositiveButtonClickListener {
+            val date = DateFormat
+                .getDateInstance(DateFormat.LONG)
+                .format(Date(it))
+            val attachment = Attachment(
+                type = "date",
+                extraData = mutableMapOf("payload" to date)
+            )
+            binding.messageInputView.submitCustomAttachments(
+                attachments = listOf(attachment),
+                viewHolderFactory = DateAttachmentPreviewFactory()
+            )
+        }
+
+        // Show the date picker dialog when the attachment button is clicked
+        binding.messageInputView.setAttachmentButtonClickListener {
+            datePickerDialog.show(requireActivity().supportFragmentManager, null)
+        }
     }
 
     private fun initMessagesViewModel() {
@@ -230,3 +275,114 @@
         return chatError.streamCode == 4
     }
 }
+
+class DateAttachmentPreviewFactory : SelectedCustomAttachmentViewHolderFactory {
+    override fun createAttachmentViewHolder(
+        attachments: List<Attachment>,
+        parent: ViewGroup,
+    ): BaseSelectedCustomAttachmentViewHolder {
+        return ItemDateAttachmentPreviewBinding
+            .inflate(LayoutInflater.from(parent.context), parent, false)
+            .let(::DateAttachmentPreviewViewHolder)
+    }
+
+    class DateAttachmentPreviewViewHolder(
+        private val binding: ItemDateAttachmentPreviewBinding,
+    ) : BaseSelectedCustomAttachmentViewHolder(binding.root) {
+
+        override fun bind(attachment: Attachment, onAttachmentCancelled: (Attachment) -> Unit) {
+            binding.dateTextView.text = attachment.extraData["payload"].toString()
+            binding.deleteButton.setOnClickListener {
+                onAttachmentCancelled(attachment)
+            }
+        }
+    }
+}
+
+class DateAttachmentFactory : AttachmentFactory {
+
+    override fun canHandle(message: Message): Boolean {
+        // Use the factory only for date attachments
+        return message.attachments.any { it.type == "date" }
+    }
+
+    override fun createViewHolder(
+        message: Message,
+        listeners: MessageListListenerContainer?,
+        parent: ViewGroup,
+    ): InnerAttachmentViewHolder {
+        // Create an inner ViewHolder with the attachment content
+        return ItemDateAttachmentBinding
+            .inflate(LayoutInflater.from(parent.context), parent, false)
+            .let { DateAttachmentViewHolder(it, listeners) }
+    }
+
+    class DateAttachmentViewHolder(
+        private val binding: ItemDateAttachmentBinding,
+        listeners: MessageListListenerContainer?,
+    ) : InnerAttachmentViewHolder(binding.root) {
+
+        private lateinit var message: Message
+
+        init {
+            // Handle clicks on the attachment content
+            binding.dateTextView.setOnClickListener {
+                listeners?.attachmentClickListener?.onAttachmentClick(
+                    message,
+                    message.attachments.first()
+                )
+            }
+            binding.dateTextView.setOnLongClickListener {
+                listeners?.messageLongClickListener?.onMessageLongClick(message)
+                true
+            }
+        }
+
+        override fun onBindViewHolder(message: Message) {
+            this.message = message
+
+            // Display the date from the attachment extras
+            binding.dateTextView.text = message.attachments
+                .first { it.type == "date" }
+                .extraData["payload"]
+                .toString()
+        }
+    }
+}
+
+class QuotedDateAttachmentFactory: QuotedAttachmentFactory {
+    override fun canHandle(message: Message): Boolean {
+        return message.attachments.any { it.type == "date" }
+    }
+
+    override fun generateQuotedAttachmentView(message: Message, parent: ViewGroup): View {
+        return QuotedDateAttachmentView(parent.context).apply {
+            showDate(message.attachments.first())
+        }
+    }
+}
+
+class QuotedDateAttachmentView(context: Context): FrameLayout(context) {
+
+    @OptIn(InternalStreamChatApi::class)
+    private val binding = ViewQuotedDateAttachemntBinding.inflate(inflater, this)
+
+    fun showDate(attachment: Attachment) {
+        binding.dateTextView.text = parseDate(attachment)
+    }
+
+    private fun parseDate(attachment: Attachment): String {
+        val date = attachment.extraData["payload"].toString()
+        return StringBuilder().apply {
+            val dateTime = SimpleDateFormat("MMMMM dd, yyyy", Locale.getDefault()).parse(date) ?: return@apply
+            val year = Calendar.getInstance().apply {
+                timeInMillis = dateTime.time
+            }.get(Calendar.YEAR)
+            if (Calendar.getInstance().get(Calendar.YEAR) != year) {
+                append(year).append("\n")
+            }
+            append(date.replace(", $year", ""))
+        }.toString()
+    }
+
+}
\ No newline at end of file
Index: stream-chat-android-ui-components-sample/src/main/res/values/dimens.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/stream-chat-android-ui-components-sample/src/main/res/values/dimens.xml b/stream-chat-android-ui-components-sample/src/main/res/values/dimens.xml
--- a/stream-chat-android-ui-components-sample/src/main/res/values/dimens.xml	(revision 627e8513cd244917129896a3258b0781a6139562)
+++ b/stream-chat-android-ui-components-sample/src/main/res/values/dimens.xml	(date 1653477618608)
@@ -25,4 +25,7 @@
 
     <dimen name="badge_horizontal_offset">8dp</dimen>
     <dimen name="badge_vertical_offset">6dp</dimen>
+
+    <dimen name="stream_ui_spacing_tiny">2dp</dimen>
+
 </resources>
Index: stream-chat-android-ui-components-sample/src/main/res/layout/item_date_attachment_preview.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/stream-chat-android-ui-components-sample/src/main/res/layout/item_date_attachment_preview.xml b/stream-chat-android-ui-components-sample/src/main/res/layout/item_date_attachment_preview.xml
new file mode 100644
--- /dev/null	(date 1653477618607)
+++ b/stream-chat-android-ui-components-sample/src/main/res/layout/item_date_attachment_preview.xml	(date 1653477618607)
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.cardview.widget.CardView xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:layout_margin="8dp"
+    app:cardBackgroundColor="#DBDBDB"
+    app:cardCornerRadius="16dp"
+    app:cardElevation="0dp">
+
+    <TextView
+        android:id="@+id/dateTextView"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_gravity="center_vertical"
+        android:layout_margin="16dp"
+        android:textColor="#000000" />
+
+    <ImageView
+        android:id="@+id/deleteButton"
+        android:layout_width="24dp"
+        android:layout_height="24dp"
+        android:layout_gravity="top|end"
+        android:layout_margin="8dp"
+        android:src="@drawable/ic_close"
+        tools:ignore="ContentDescription" />
+
+</androidx.cardview.widget.CardView>
\ No newline at end of file
Index: stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/application/App.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/application/App.kt b/stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/application/App.kt
--- a/stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/application/App.kt	(revision 627e8513cd244917129896a3258b0781a6139562)
+++ b/stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/application/App.kt	(date 1653477618607)
@@ -24,8 +24,14 @@
 import coil.decode.ImageDecoderDecoder
 import io.getstream.chat.android.client.utils.internal.toggle.ToggleService
 import io.getstream.chat.android.core.internal.InternalStreamChatApi
+import io.getstream.chat.android.ui.ChatUI
+import io.getstream.chat.android.ui.message.list.adapter.viewholder.attachment.AttachmentFactoryManager
+import io.getstream.chat.android.ui.message.list.adapter.viewholder.attachment.DefaultQuotedAttachmentMessageFactory
+import io.getstream.chat.android.ui.message.list.adapter.viewholder.attachment.QuotedAttachmentFactoryManager
 import io.getstream.chat.ui.sample.data.user.SampleUser
 import io.getstream.chat.ui.sample.data.user.UserRepository
+import io.getstream.chat.ui.sample.feature.chat.DateAttachmentFactory
+import io.getstream.chat.ui.sample.feature.chat.QuotedDateAttachmentFactory
 
 class App : Application() {
 
@@ -49,6 +55,12 @@
         )
         ApplicationConfigurator.configureApp(this)
         initializeToggleService()
+
+        ChatUI.attachmentFactoryManager = AttachmentFactoryManager(listOf(DateAttachmentFactory()))
+        ChatUI.quotedAttachmentFactoryManager = QuotedAttachmentFactoryManager(listOf(
+            QuotedDateAttachmentFactory(),
+            DefaultQuotedAttachmentMessageFactory()
+        ))
     }
 
     private fun getApiKey(): String {
```

</details>

How to test:

1. Run on this branch, login as Filip, go to Real Window and find the attachments as on image above
2. Apply the patch, now clicking on the attachment icon will prompt you to enter a date
3. Send some messages with dates and quote them and see if they render correctly

#### General
- [x] Assigned a person / code owner group (required)
- [ ] Thread with the PR link started in a respective Slack channel (#android-chat-core or #android-chat-ui) (required)
- [x] PR targets the `develop` branch
- [x] PR is linked to the GitHub issue it resolves

#### Code & documentation
- [x] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [x] Comparison screenshots added for visual changes
- [x] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### 🎉 GIF

giphy

### CODE REVIEW

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/view/internal/MessageReplyView.kt [hunk 1]
```diff
@@ -71,9 +69,14 @@ internal class MessageReplyView : FrameLayout {
...
 
+    /**
+     * @param message [Message] that was replied to.
+     * @param isMine If the message containing the reply was current users or not.
+     * @param style The style to be applied to the view.
+     */
     fun setMessage(message: Message, isMine: Boolean, style: MessageReplyStyle?) {
         setUserAvatar(message)
-        setAvatarPosition(isMine)
+        setAvatarPosition(message.isMine())
```
Comment (filbabic) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/view/internal/MessageReplyView.kt:79]: do we need this change?

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/view/internal/MessageReplyView.kt [hunk 2]
```diff
@@ -120,16 +123,16 @@ internal class MessageReplyView : FrameLayout {
...
+     * @param isMine Whether the message containing the reply is from the current user or not.
      * @param style [MessageReplyStyle] contains the styles of the background.
      */
     private fun setReplyBackground(message: Message, isMine: Boolean, style: MessageReplyStyle?) {
         val shapeAppearanceModel = ShapeAppearanceModelFactory.create(
             context,
             REPLY_CORNER_RADIUS,
             0f,
-            isMine,
+            message.isMine(),
```
Comment (filbabic) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/view/internal/MessageReplyView.kt:135]: same here

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/view/internal/MessageReplyView.kt [hunk 3]
```diff
@@ -144,10 +147,13 @@ internal class MessageReplyView : FrameLayout {
                     }
                     setTint(color)
                 }
-                isMine -> {
+                message.isMine() -> {
```
Comment (filbabic) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/view/internal/MessageReplyView.kt:150]: and here

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/view/internal/MessageReplyView.kt [hunk 4]
```diff
@@ -144,10 +147,13 @@ internal class MessageReplyView : FrameLayout {
...
                     paintStyle = Paint.Style.FILL_AND_STROKE
-                    val color =
+                    val color = if (isMine) {
```
Comment (filbabic) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/view/internal/MessageReplyView.kt:152]: This condition is a bit weird - we check if hte message is mine and then proceed to check again for the BG color?

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/view/internal/MessageReplyView.kt [hunk 5]
```diff
@@ -168,18 +174,11 @@ internal class MessageReplyView : FrameLayout {
...
         } else {
-            when (attachment.type) {
-                ModelType.attach_file -> showFileTypeLogo(attachment.mimeType)
-                ModelType.attach_image -> showAttachmentThumb(attachment.imagePreviewUrl)
-                ModelType.attach_giphy,
-                ModelType.attach_video,
-                -> showAttachmentThumb(attachment.thumbUrl)
-                else -> showAttachmentThumb(attachment.image)
-            }
+            binding.attachmentContainer.isVisible = false
```
Comment (filbabic) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/view/internal/MessageReplyView.kt:181]: In the previous condition, we set the `logoContainer` visibility to false and here we change the attachment container - any reason why this change?

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/viewholder/attachment/QuotedAttachmentFactoryManager.kt [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    ) {
+        val quotedAttachmentFactory = quotedAttachmentFactories.firstOrNull { it.canHandle(message) }
+        val view = quotedAttachmentFactory?.generateQuotedAttachmentView(message, parent)
+            ?: ChatUI.attachmentFactoryManager.createViewHolder(message, null, parent).apply {
+                onBindViewHolder(message)
+            }.itemView
+
+        parent.removeAllViews()
+        parent.addView(view)
+        parent.invalidate()
```
Comment (filbabic) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/viewholder/attachment/QuotedAttachmentFactoryManager.kt:60]: does invalidating have any performance impact?

File: stream-chat-android-ui-components/src/main/res/layout/stream_ui_message_reply_view.xml [hunk 7]
```diff
@@ -45,47 +45,30 @@
         >
 
         <FrameLayout
-            android:id="@+id/logoContainer"
+            android:id="@+id/attachmentContainer"
```
Comment (filbabic) [inline stream-chat-android-ui-components/src/main/res/layout/stream_ui_message_reply_view.xml:48]: Okay this explains the logoContainer change

File: stream-chat-android-ui-components/src/main/res/values/styles.xml [hunk 8]
```diff
@@ -353,7 +353,7 @@
         <item name="streamUiMessageTextStyleThreadSeparator">normal</item>
 
         <!-- Message Reply background color -->
-        <item name="streamUiMessageReplyBackgroundColorMine">@color/stream_ui_bars_background</item>
+        <item name="streamUiMessageReplyBackgroundColorMine">@color/stream_ui_grey_gainsboro</item>
```
Comment (filbabic) [inline stream-chat-android-ui-components/src/main/res/values/styles.xml:356]: Any reason for the color change?

**Review by filbabic — CHANGES_REQUESTED** [review]
Left some comments - other than that works and looks great! :]

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/view/internal/MessageReplyView.kt (hunk 1 above)
Comment (TGazica) [reply stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/view/internal/MessageReplyView.kt:79]: yes, if you look at the description, isMine is the bool of the message containing the replay, where as message is the quoted message. We need to check against the quoted message to show the avatar on the right side.

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/view/internal/MessageReplyView.kt (hunk 2 above)
Comment (TGazica) [reply stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/view/internal/MessageReplyView.kt:135]: Same reasoning as above, just for the shape.

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/view/internal/MessageReplyView.kt (hunk 3 above)
Comment (TGazica) [reply stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/view/internal/MessageReplyView.kt:150]: and here as well, to show the appropriate background colour for the quoted message.

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/view/internal/MessageReplyView.kt (hunk 2 above)
Comment (filbabic) [reply stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/view/internal/MessageReplyView.kt:135]: Sounds good - my bad!

File: stream-chat-android-ui-components/src/main/res/layout/stream_ui_message_reply_view.xml (hunk 7 above)
Comment (TGazica) [reply stream-chat-android-ui-components/src/main/res/layout/stream_ui_message_reply_view.xml:48]: Yup, it becomes just the root for the attachment.

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/view/internal/MessageReplyView.kt (hunk 5 above)
Comment (TGazica) [reply stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/view/internal/MessageReplyView.kt:181]: As mentioned down below, it is just a container where the attachment is populated.

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/view/internal/MessageReplyView.kt (hunk 4 above)
Comment (TGazica) [reply stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/view/internal/MessageReplyView.kt:152]: this is connected to the colour change down below. Before there was a bug where the avatar side was not determined properly. Now i changed that the users reply has the same colour as his messages so when someone else replies to current users message the MessageReply is grey. If we set the colour as you commented back, we then need to show the border, otherwise there is no UI distinction for the quoted message. eg: 
Screenshot 2022-05-26 at 11 21 57

And if we change the colour down below to match the regular message colour then invert of the message above applies, grey background for reply inside grey message.

The check is as follows: message.isMine for the replied to message and is mine for the message containing the reply. If bot are true set different background for replied to message.

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/viewholder/attachment/QuotedAttachmentFactoryManager.kt [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+        return quotedAttachmentFactories.any { it.canHandle(message) } ||
+            ChatUI.attachmentFactoryManager.canHandle(message)
+    }
+
+    /**
+     * Create a view for the quoted attachments.
+     *
+     * @param message The message containing attachments that we are going to render.
+     * @param parent The parent View where the attachment content view is supposed to be placed.
+     * @return A quoted attachment view to be placed inside the quoted message.
```
[the remark this answers is not in the document]
Comment (TGazica) [reply stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/viewholder/attachment/QuotedAttachmentFactoryManager.kt:46]: I changed the implementation while working on it and forgot to remove return docs, will refactor the function name.

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/viewholder/attachment/QuotedAttachmentFactoryManager.kt (hunk 6 above)
Comment (TGazica) [reply stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/viewholder/attachment/QuotedAttachmentFactoryManager.kt:60]: could have, but it is not necessary to call it, removing it.

File: stream-chat-android-ui-components/src/main/res/values/styles.xml (hunk 8 above)
Comment (TGazica) [reply stream-chat-android-ui-components/src/main/res/values/styles.xml:356]: Take a look at the comment above with the image reply.

**Review by filbabic — APPROVED** [review]
### ☑️Reviewer Checklist
- [x] UI Components sample runs & works
- [x] UI Changes correct (before & after images)
- [x] New feature tested and works
- [x] Release notes and docs clearly describe changes
- [x] All code we touched has new or updated KDocs
